app/models/fact_doi.rb

```diff
@@ -1,26 +1,44 @@
 class FactDoi
   def info(doi)
-    json = fetch(doi)
-    return if json == 'Error'
+    external_data = fetch(doi)
+    return if external_data == 'Error'
 
+    metadata = extract_metadata(external_data)
+    metadata[:doi] = doi
+    metadata[:open_url] = open_url(metadata)
+    metadata
+  end
+
+  def extract_metadata(external_data)
     {
-      title: json['message']['title'],
-      publisher: json['message']['publisher']
+      genre: external_data['genre'],
+      title: external_data['title'],
+      year: external_data['year'],
+      publisher: external_data['publisher'],
+      oa: external_data['is_oa'],
+      oa_status: external_data['oa_status'],
+      best_oa_location: external_data['best_oa_location'],
+      journal_issns: external_data['journal_issns'],
+      journal_name: external_data['journal_name']
     }
   end
 
   def url(doi)
-    "https://api.crossref.org/works/#{doi}"
+    "https://api.unpaywall.org/v2/#{doi}?email=timdex@mit.edu"
   end
 
   def fetch(doi)
     resp = HTTP.headers(accept: 'application/json').get(url(doi))
     if resp.status == 200
       JSON.parse(resp.to_s)
     else
-      Rails.logger.debug("Fact lookup error. DOI #{doi} detected but crossref returned no data")
+      Rails.logger.debug("Fact lookup error. DOI #{doi} detected but unpaywall returned no data")
       Rails.logger.debug("URL: #{url(doi)}")
       'Error'
     end
   end
+
+  def open_url(metadata)
```

Comment:
Non-blocking, but I think OpenURL is one word.

Reply:
Good catch. Looking at this with fresh eyes this morning and with that comment in mind, I think it may be better named something more like "primo_url" or "link_resolver_url"

Reply:
After further discussion on slack, I went with `link_resolver_url`. Thanks for the input!